realm/realm-library/src/main/cpp/io_realm_RealmFileUserStore.cpp

```diff
@@ -31,11 +31,15 @@ Java_io_realm_RealmFileUserStore_nativeGetCurrentUser (JNIEnv *env, jclass)
 {
     TR_ENTER()
     try {
-
-        const std::shared_ptr<SyncUser> &user = SyncManager::shared().get_current_user();
-        if (user) {
-            return to_jstring(env, user->refresh_token().data());
-        } else {
+        try {
+            const std::shared_ptr<SyncUser> &user = SyncManager::shared().get_current_user();
+            if (user) {
+                return to_jstring(env, user->refresh_token().data());
+            } else {
+                return nullptr;
+            }
+        } catch (std::logic_error e) {
+            ThrowException(env, IllegalState, "Multiple users have been logged in. `currentUser` is only valid if one single user is logged in.");
```

Comment:
I just feel really bad when we need to convert exceptions here and there. Or shall we use a different error class in OS in this case? (maybe just subclass the `logic_error` and throw it.) Then we still can have a centralized place to do exception converting.